apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,195 @@
 // @flow strict
 
 import * as React from 'react';
+import { View } from 'react-native';
+import * as DateFNS from 'date-fns';
 import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
-import { Text, StyleSheet } from '@kiwicom/universal-components';
+import { StyleSheet } from '@kiwicom/universal-components';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import {
+  ConnectionCard,
+  type TripSectorWithId,
+} from '@kiwicom/margarita-components';
 
 import type { ResultsListItem as ResultsListItemType } from './__generated__/ResultsListItem.graphql';
 
 type Props = {|
   +data: ResultsListItemType,
 |};
 
-const ResultListItem = (props: Props) => {
-  if (props.data == null) {
+export type Route = $PropertyType<ResultsListItemType, 'route'>;
+export type RouteItem = $ElementType<$NonMaybeType<Route>, number>; // number because arrays are number-indexed
+
+// @TODO get from config
+const dateFormat = 'ddd D MMM';
+
+class ResultListItem extends React.Component<Props> {
+  parseDate = (date: ?Date) => {
+    if (date) return DateFNS.parse(date);
     return null;
-  }
-  const { flyFrom, flyTo, localDeparture, localArrival, price } = props.data;
-  /**
-   * TODO: properly handle possible undefined props in final version of list
-   * (for example: should be entry with undefined price still displayed?)
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
-};
+  };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem) => {
```

Comment:
Could you replace `route` with `routeItem`, otherwise, it's confusing with the naming of the next function.

Reply:
OK